File: K47 User Manual/2_light_and_infrared_sensors/13_lightBlock/Code/Python/lightBlock.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import time

LightBreakPin = 11
LedPin = 16

def setup():
	GPIO.setmode(GPIO.BOARD)       # Numbers GPIOs by physical location
	GPIO.setup(LedPin, GPIO.OUT)   # Set LedPin's mode is output
	GPIO.setup(LightBreakPin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
	GPIO.output(LedPin, GPIO.LOW) # Set LedPin low to off led


def loop():
	while True:
		if(GPIO.input(LightBreakPin) == 0):
			print 'Be covered....'
			GPIO.output(LedPin, GPIO.HIGH)
		else:
			GPIO.output(LedPin, GPIO.LOW)
		time.sleep(0.2)
def destroy():
	GPIO.output(LedPin, GPIO.LOW)     # led off
	GPIO.cleanup()                     # Release resource

if __name__ == '__main__':     # Program start from here
	setup()
	try:
		loop()
	except KeyboardInterrupt:  # When 'Ctrl+C' is pressed, the child program destroy() will be  executed.
		destroy()

File: K47 User Manual/2_light_and_infrared_sensors/13_lightBlock/Code/C/lightBlock.c
#include <wiringPi.h>
#include <stdio.h>

#define LightBreakPin     0
#define LedPin            4

int main(void)
{
	if(wiringPiSetup() == -1)
	{
		printf("setup wiringPi failed !");
		return -1; 
	}
	
	pinMode(LightBreakPin, INPUT);
	pullUpDnControl(LightBreakPin, PUD_UP);
	pinMode(LedPin,  OUTPUT);

	while(1)
	{
		if(digitalRead(LightBreakPin) == LOW)
		{
			printf("Be covered....\n");
			digitalWrite(LedPin, HIGH);     //led on
		}	
		else
		{
			digitalWrite(LedPin, LOW);		//led off	
		}
	}

	return 0;
}
